frontends/ol-components/src/components/LearningResourceExpanded/LearningResourceExpanded.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { LearningResourceExpanded } from "./LearningResourceExpanded"
+import type { LearningResourceExpandedProps } from "./LearningResourceExpanded"
+import { ResourceTypeEnum } from "api"
+import { factories } from "api/test-utils"
+import { ThemeProvider } from "../ThemeProvider/ThemeProvider"
+import { getReadableResourceType } from "ol-utilities"
+import invariant from "tiny-invariant"
+import type { LearningResource } from "api"
+
+const IMG_CONFIG: LearningResourceExpandedProps["imgConfig"] = {
+  key: "fake-key",
+  width: 385,
+  height: 200,
+}
+
+const setup = (resource: LearningResource) => {
+  return render(
+    <BrowserRouter>
+      <LearningResourceExpanded resource={resource} imgConfig={IMG_CONFIG} />
+    </BrowserRouter>,
+    { wrapper: ThemeProvider },
+  )
+}
+
+describe("Learning Resource Expanded", () => {
+  it.each(
+    Object.values(ResourceTypeEnum).filter(
+      (type) => type !== ResourceTypeEnum.Video,
+    ),
+  )('Renders image, title and link for resource type "%s"', (resourceType) => {
+    const resource = factories.learningResources.resource({
+      resource_type: resourceType,
+    })
+
+    setup(resource)
+
+    const images = screen.getAllByRole("img")
+    const image = images.find((img) =>
+      img
+        .getAttribute("src")
+        ?.includes(encodeURIComponent(resource.image?.url ?? "")),
+    )
+    expect(image).toBeInTheDocument()
+    invariant(image)
+    expect(image).toHaveAttribute("alt", resource.image?.alt ?? "")
+
+    screen.getByRole("heading", { name: resource.title })
+
+    const linkName =
+      resource.resource_type === ResourceTypeEnum.Podcast
+        ? `Listen to ${getReadableResourceType(resource.resource_type)}`
+        : `Take ${getReadableResourceType(resource.resource_type)}`
+
+    if (linkName) {
+      const link = screen.getByRole("link", {
+        name: linkName,
+      }) as HTMLAnchorElement
+      expect(link.href).toBe(`${resource.url}/`)
+    }
+  })
+
+  it(`Renders card and title for resource type "${ResourceTypeEnum.Video}"`, () => {
+    const resource = factories.learningResources.resource({
+      resource_type: ResourceTypeEnum.Video,
+    })
+
+    setup(resource)
+    // eslint-disable-next-line testing-library/no-node-access
+    const embedlyCard = document.querySelector(".embedly-card")
+    invariant(embedlyCard)
+    expect(embedlyCard).toHaveAttribute("href", resource.url)
+
+    screen.getByRole("heading", { name: resource.title })
+  })
+
+  it.each(
+    Object.values(ResourceTypeEnum).filter(
+      (type) => type !== ResourceTypeEnum.Video,
+    ),
+  )('Renders topic section for resource type "%s"', (resourceType) => {
+    const resource = factories.learningResources.resource({
+      resource_type: resourceType,
+    })
+
+    setup(resource)
+
+    const section = screen
+      .getByRole("heading", { name: "Topics" })!
+      // eslint-disable-next-line testing-library/no-node-access
+      .closest("section")!
```

Comment:
Is there a better alternative here for finding a heading, then targeting the parent section for tests against its contents?

Reply:
I think this is the best way. Playwright does have nicer locators for this sort of thing.

We can globally disable ` eslint-disable-next-line testing-library/no-node-access`. I think it's been annoying.